src/java.base/share/classes/java/util/ServiceLoader.java

```diff
@@ -1233,6 +1238,7 @@ private Class<?> nextProviderClass() {
         }
 
         @SuppressWarnings("unchecked")
+        @SideEffectsOnly("this")
```

Comment:
Is this annotation true?

I see that [this method call](https://github.com/typetools/jdk/blob/49144c9e4afdc065a6029b5e251aac97d263167d/src/java.base/share/classes/java/util/ServiceLoader.java#L679) is invoked from this method, which appears to mutate something beyond `this`.

I only caught this because `hasNextService()` is named differently from `hasNext()` methods, for which the `@SideEffectsOnly("this")` annotation is almost invariably true.

Reply:
Thanks for catching that.